tests/StorageApiTestCase.php

```diff
@@ -673,7 +685,140 @@ function (self $that, array $config, $useExistingBranch = false) {
         ];
     }
 
-    private function createOrReuseDevBranch(self $that, $useExistingBranch = false)
+    /**
+     * Usage:
+     *  - `$getClient($this)` -> return default client which use branch aware client for calling default branch
+     *  - `$getClient($this, [...])` -> return client with custom config which use branch aware client for calling default branch
+     *  - `$getClient($this, [], true)` -> return default client which use same branch as before
+     *  - `$getClient($this, [...], true)` -> return client with custom which use same branch as before
+     *
+     * @see provideComponentsClient
+     */
```

Comment:
Už jsem to psal v tom minulém PR, že mi ta closure nepřijde dobré řešení a jsme tu. Mám random dataprovider, který vrací random dvě různé closure s různými parametry. IMHO to měla být nějaká třída s jasným interfacem :(

Reply:
Ale nedává smysl to tím tady teď brzdit.